src/cage_keeper.py

```diff
@@ -205,20 +214,20 @@ def facilitate_processing_period(self):
 
         # Cage all ilks
         for ilk in ilks:
-            self.dss.end.cage(ilk).transact(gas_price=self.gas_price())
+            self.dss.end.cage(ilk).transact(gas_price=self.gas_price)
```

Comment:
So, all the changed to `gas_price` here just use the default or dynamic gas model from init.  But that is only set once on startup.  Do we not want all these calls to dynamically adjust gas on every call?  Is there a way we can even do something simple like make `gas_price = DynamicGasPrice` or `gas_price = DefaultGasPrice` above, assigning the function to this variable, and then make all of these calls `gas_price()` again, but with the variable pointing at the appropriate model?  Obviously we have to figure out how to pass the API key in the former case, but then each call would get an appropriate gas amount.

Reply:
Apologies for the belated response. 

The `DynamicGasPrice` class inherits `GasPrice` [source](https://github.com/makerdao/pymaker/blob/master/pymaker/gas.py#L21), which contains `get_gas_price()` that is responsible for returning a gas price at a specified time. Calling `get_gas_price()` is handled by the `pymaker.Transact` object [here](https://github.com/makerdao/pymaker/blob/95a40eba846a1ed4abd783b526df98a1369ea197/pymaker/__init__.py#L612)